### PR TITLE
tf2.0

### PR DESCRIPTION
**Pull request by arnavdas88**

### CODE REVIEW

File: src/TensorFlowNET.Core/tf2.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+******************************************************************************/
+
+using System;
+
+namespace Tensorflow
+{
+    public static class tf2
```
Comment (Oceania2018) [inline src/TensorFlowNET.Core/tf2.cs]: What's this `static class tf2` for? Should it be merged into `public partial class tensorflow { }`?

File: src/TensorFlowNET.Core/Binding.Util.cs [hunk 2]
```diff
@@ -333,5 +333,30 @@ public static bool isinstance(object Item1, object tuple)
...
+                var objProp = objProps_w_IndexParams[i];
+                var objPropiParam = objProp.GetIndexParameters();
+                ret.Add(objProp.Name, objProp.GetValue(obj, new object[]{0}));
+            }
+            for(int i = 0; i < objFields.Length; i++)
+                ret.Add(objFields[i].Name, objFields[i].GetValue(obj));
+            return ret;
+            */
+        }
+        public static long id(object obj)
```
Comment (Nucs) [inline src/TensorFlowNET.Core/Binding.Util.cs]: Can I have reference to the original method?

Comment (arnavdas88) [reply src/TensorFlowNET.Core/Binding.Util.cs]: the original method is from core python i guess.... it gives the unique id associated to any variable

File: src/TensorFlowNET.Core/tf2.cs (hunk 1 above)
Comment (arnavdas88) [reply src/TensorFlowNET.Core/tf2.cs]: actually tf2 is a class that is from tf2.0 and it cant be merged coz it is a different distinct class....

**Comment by arnavdas88** [comment]
And @Oceania2018 , @Nucs  , Can you guys give me a little insight about https://github.com/tensorflow/tensorflow/blob/25006be096cd4f0242a3b979fb212bbd192127b3/tensorflow/python/module/module.py#L332 , here vars() keyword has been used.... but I don't know what data does it extract. just the variables or the functions too???

**Comment by Nucs** [comment]
image
vars should return `Dictionary<string, object>`, it seems to access `module.__dict__` and return its content ([read more here](https://codesachin.wordpress.com/2016/06/09/the-magic-behind-attribute-access-in-python/)).
C#'s equivalent would be to extract all public fields or properties depending on how you plan to do these in classes that inherit Module

File: src/TensorFlowNET.Core/Binding.Util.cs (hunk 2 above)
Comment (Nucs) [reply src/TensorFlowNET.Core/Binding.Util.cs]: I can't find an equivalent in python, can you write here a small code that calls `id`?

**Comment by Nucs** [comment]
@arnavdas88 I'm guessing you'll keep pushing to this PR untill you finish your work right? If so, let us know when you need us by requesting reviews.

**Comment by arnavdas88** [comment]
> @arnavdas88 I'm guessing you'll keep pushing to this PR untill you finish your work right? If so, let us know when you need us by requesting reviews.

Awsome...

**Pull request closed without merging**